hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -181,6 +182,25 @@ public void incrementCount(List<Long> txIDs)
     }
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   */
+  @Override
+  public int resetCount(List<Long> txIDs) throws IOException, TimeoutException {
+    lock.lock();
+    try {
+      if (txIDs == null || txIDs.isEmpty()) {
+        txIDs = getFailedTransactions().stream()
+            .map(DeletedBlocksTransaction::getTxID)
+            .collect(Collectors.toList());
+      }
+      return deletedBlockLogStateManager.resetRetryCountOfTransactionInDB(
+          new ArrayList<>(new HashSet<>(txIDs)));
```

Comment:
resetRetryCountOfTransactionInDB can take List as input. Then we dont need the conversions.

Reply:
> resetRetryCountOfTransactionInDB can take List as input. Then we dont need the conversions.

Here the first conversion to ArrayList is essential, coz the list itself is an interface and cannot be instantiated when triggering @Replicate method,  the argument needs to be instantiated. 
The second conversion is to eliminate the duplicate txns, this also helps reduce further processes and data transfer if send too much same txId. I think it also meaningful.